File: codes/lua/inheritance/abst/person.lua
local Person = {}
Person.__index = Person

function Person.new()    -- The constructor
    local instance = {}
    setmetatable(instance, Person)
    return instance
end

function Person:info()
    print("====INFO====");
    self:info2()
end

function Person:info2()
    error("Cannot call abstract method")
end

return Person

File: codes/javascript/inheritance/abst/es6/person.js
class Person {
    constructor() {
        if (this.constructor == Person) {
            throw new Error("Cannot instanciate abstract class");
        }
    }

    info() {
        console.log("====INFO===");
        this.info2();
    }

    info2(){
        throw new Error("Cannot call abstract method");
    }
}

module.exports = Person;
